README.md

```diff
@@ -110,6 +110,21 @@ Like node.js, callbacks for methods that accept them are optional but suggested
 you omit the callback, errors will be thrown as exceptions). The first callback parameter is
 reserved for passing errors. It will be `null` if no errors occurred and should always be checked.
 
+#### Support for Promises is here!
+Promise based API mimics the way Node [implements](https://nodejs.org/api/fs.html#fs_fs_promises_api) them. Both `Shell` and `FileSystem` now have `promises` object attached alongside regular callback style methods. Method names are identical to their callback counterparts with the difference that instead of receiving third argument as a callback, they return a Promise that is resolved or rejected based on the success of method execution.
+> Please note that `exists` method will always throw, since it was [deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback) by Node.
+
+See example below:
+
+```javascript
+const fs = new Filer.FileSystem().promises;
+fs.open('/myfile', 'w+')
+  .then(fd => fs.close(fd))
+  .then(() => fs.stat('/myfile'))
```

Comment:
Let's do two things here:

1. Let's also show the non-promises way of doing this, so people can compare the difference.  That is, let's have two code blocks and call out the difference between the two so people can clearly understand what you're doing here with respect to the original API.

2. Let's consider using `fstat` above, and closing the file descriptor toward the end, so the example is less contrived.  So the order would be open, fstat, close.

Reply:
I used example from the Overview section to keep it consistent. I could just link my example to the Overview or re-write it completely as you suggested using `fstat.`

I am inclined to go with the first option of mere linking, since it will reduce clutter of an already code-rich readme. Let me know what you think.

Reply:
We can still indicate the differences between styles while linking the examples.